### ISSUE TITLE
Provide access to structured REST errors that are returned

### ISSUE THREAD

**Issue by element-of-surprise** [Azure/azure-kusto-go#12]
Originally a part of issue: https://github.com/Azure/azure-kusto-go/issues/10

Errors returned as part of Kusto can come in a variety of shapes.  Some are errors generated by authorization systems, some are generated by Kusto, etc....

These errors may come in two forms

- Standard string
- JSON encoded

The JSON encoded may also come in different forms.  Some are OneAPI errors, but off an earlier spec that is no longer valid or some other JSON encoded structure.

Currently, we simply try to look at the OneAPI errors we have seen and simply put them as wrapped errors inside Go's error type.  This probably has limited use.

Go has a wide variety of error detection methods.  However, these are all predicated on that the Go code knows what the errors are going to be.  At this time, I don't think it is realistic to believe we can do that.  We simply understand the severity of the issue and where the error comes from.  

In those issues, we have error codes that can be introspected via the errors library.  

But we do not have anything to allow dynamic introspection of the data when it is JSON encoded.  This is being asked for.

Here is the original part of the request:

```
Additionally, the main issue is that the actual Kusto error response is basically lost. All that's returned from a Kusto call is an HTTP Error with a text string containing the http response body embedded inside it.

It appears as though Kusto returns a json struct for the error with code, message, @type, @context, etc. Shouldn't there be a formal Kusto error struct that's optionally set and returned with errors returned by calls to Kusto Go APIs ?
```

**Comment by vladikbr** [comment]
Are you going to make other enhancements here or should we just close this?

**Comment by element-of-surprise** [comment]
I'm considering a new errors package in the future.  The OneError thing is like a moving target and the backend sometimes sends them and sometimes doesn't.  I'm thinking of doing something simpler like I implemented in MSAL.

I think its fine to close this and I can open up another one that has something more well thought out than my one line comment above.

**Comment by AsafMah** [comment]
Finally fixed in 1.0.0-preview
